### ISSUE TITLE
pyoomph: get_id_string is weird

### ISSUE THREAD

**Issue by Akuli** [Akuli/oomph#106]
Ideally, it would behave like this:
- Generic types (currently just List ~and Optional~): name and string of generic arg
- Auto types: error
- Union types: what is being unioned, sorted in some consistent way (after #105)

Needed to complete #70

**Comment by Akuli** [comment]
#111 will improve this slightly, but won't fix it completely
